crates/next-core/src/next_manifests/client_reference_manifest.rs

```diff
@@ -202,7 +202,17 @@ impl ClientReferenceManifest {
                             // It's possible that a chunk also emits CSS files, that will
                             // be handled separatedly.
                             .filter(|path| path.ends_with(".js"))
-                            .map(|path| format!("{}{}", path, suffix_path))
+                            .map(|path| {
+                                format!(
+                                    "{}{}{}",
+                                    prefix_path,
+                                    path.split('/')
+                                        .map(urlencoding::encode)
+                                        .collect::<Vec<_>>()
+                                        .join("/"),
```

Comment:
You could use https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.format here to avoid the two allocations (Vec, String)

Reply:
We do the exact same thing here as well BTW: https://github.com/vercel/next.js/blob/23681508ca34b66a6ef55965c5eac57de20eb67f/turbopack/crates/turbo-tasks-fs/src/util.rs#L160-L163